docs/api.md

```diff
@@ -2168,6 +2169,11 @@ puppeteer.launch().then(async browser => {
 
 [ConsoleMessage] objects are dispatched by page via the ['console'](#event-console) event.
 
+#### consoleMessage.additionalInfo()
```

Comment:
Why not just put the url and line number on the console message itself?

Reply:
you mean extending the api? same as `type` and `args`?

Reply:
yes

Reply:
as I commented in the [origin issue](https://github.com/GoogleChrome/puppeteer/issues/3029#issuecomment-427738035):
> This is under the assumption that additionalInfo is a subset of LogEntry optional params, thus future requests for additional data can easily added

makes sense?

Reply:
also the fact the lineNumber and url are optionals, made me feel less comfortable in adding a dedicated API

Reply:
Having `additionalInfo` raises some difficult questions. Which values are considered core, and which are considered additional? Can additional info be null, or will it be an empty object? Even if we have reasoning, its not obvious to a user of the API. They would have to check the documentation every time.

That being said, it might make sense to group line, column, and url under a `location` object if they are often all null or all present.

Reply:
done. `location` it is.